crates/core_arch/src/x86/avx.rs

```diff
@@ -405,16 +405,12 @@ pub unsafe fn _mm256_div_pd(a: __m256d, b: __m256d) -> __m256d {
 /// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm256_round_pd)
 #[inline]
 #[target_feature(enable = "avx")]
-#[cfg_attr(test, assert_instr(vroundpd, b = 0x3))]
-#[rustc_args_required_const(1)]
+#[cfg_attr(test, assert_instr(vroundpd, B = 0x3))]
+#[rustc_legacy_const_generics(1)]
 #[stable(feature = "simd_x86", since = "1.27.0")]
-pub unsafe fn _mm256_round_pd(a: __m256d, b: i32) -> __m256d {
-    macro_rules! call {
-        ($imm8:expr) => {
-            roundpd256(a, $imm8)
-        };
-    }
-    constify_imm8!(b, call)
+pub unsafe fn _mm256_round_pd<const B: i32>(a: __m256d) -> __m256d {
+    static_assert_imm8!(B);
```

Comment:
This should be `imm4`. Not your fault, the original code was invalid in the first place.

Reply:
Do we also want to rename the parameter to match the intel intrinsics guide ? (it should be `rounding` here)

Reply:
Yes that's probably a good idea.

Reply:
Ok. I've pushed both changes